.github/workflows/main.yaml

```diff
@@ -0,0 +1,37 @@
+name: PR Check
+
+on:
+  push:
+    branches:
+    - "main"
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: generate-tag                                                                 
```

Comment:
Rather than generating a tag, we could use `edge` for tracking `main`. That's how other projects in the OPA or does it.

It would be good if the this flow worked when an actual tag was pushed too, and used that if found.

Reply:
What do you mean by using `edge` here? Are there any good examples in the other projects here?
I can rewrite it to only trigger on tag pushes if that is your preferred way? Like https://github.com/open-policy-agent/opa-envoy-plugin/blob/main/.github/workflows/post-tag.yaml?

Reply:
Yep! Or https://github.com/open-policy-agent/opa/blob/main/.github/workflows/post-merge.yaml#L139

I.e. use the tag `edge` for tracking what's currently the top of the main branch. That workflow should run on all pushes to main. Then, ideally we'd have another similar workflow that runs only on pushed tags, and which uses the tag for... well, tagging :)

Reply:
Ok, If I understood you correctly, the PR should be updated with two workflows. Does it look OK to you?
Happy to help getting this the way you want :+1: